### DISCUSSION TITLE
Is it possible to set a layout for a directory (and it's subdirs)?

### DISCUSSION THREAD

**Discussion by nobkd** [nuxt/content#1393] [Q&A] [answered]
I wanted to set layouts per directory (and therefore also the subdirectories), without putting it into the frontmatter of every file.

---

Ideas i had how it could work, if it's not already possible:
* by using a document like `_layout.yml` in the directory (or somehow included in the `_dir.yml`)
* by specifying the layout for routes in some combined config file

Frontmatter layouts should always override directory layouts

**Comment by nozomuikuta** [comment, marked as answer]
You could define a function on [`content:file:afterParse`](https://content.nuxtjs.org/api/advanced#contentfileafterparse) hook to set `layout` property to files based on their path like below:

```typescript
// server/plugins/content.ts

export default defineNitroPlugin((nitroApp) => {
  nitroApp.hooks.hook('content:file:afterParse', (file) => {
    if (file._id.endsWith('.md')) {
      // some logic to detect file's path
      file.layout = layoutForDir || file.layout
    }
  })
})
```

```typescript
// nuxt.config.ts

export default defineNuxtConfig({
  nitro: {
    plugins: ['~/server/plugins/content.ts']
  }
})
```

**Reply by nobkd** [reply]
That helped a lot. Thank you for the starting point.
